### PR TITLE
Geimaj/lens follow up

### PR DESCRIPTION
**Pull request by Geimaj**
adds more lens songs and gets audio from `media` rather than `animation_url`

### CODE REVIEW

File: src/types/media.ts [hunk 1]
```diff
@@ -7,6 +7,13 @@ export enum MimeEnum {
   jpeg = 'image/jpeg',
   png = 'image/png',
   pdf = 'application/pdf',
+  quicktime = 'video/quicktime'
+}
+
+export type MediaMetadata = {
```
Comment (musnit) [inline src/types/media.ts:13]:
```suggestion
export type LensMediaMetadata = {
```

File: src/types/nft.ts [hunk 2]
```diff
@@ -145,3 +146,25 @@ export const getTraitType = (nft: NFT, value: string) => {
   }
   return traitAttribute.trait_type;
 };
+
+export const getMedia = (nft: NFT, mimeTypes: MimeEnum[]) => {
```
Comment (musnit) [inline src/types/nft.ts:150]:
```suggestion
export const getLensMedia = (nft: NFT, mimeTypes: MimeEnum[]) => {
```

File: src/types/fieldExtractor.ts [hunk 3]
```diff
@@ -105,6 +107,7 @@ const titleExtractors: TitleExtractorMapping = {
 const audioUrlExtractors: AudioUrlExtractorMapping = {
   [AudioUrlExtractorTypes.METADATA_ANIMATION_URL]: (nft: NFT) => cleanURL(nft.metadata.animation_url),
   [AudioUrlExtractorTypes.ATTRIBUTES_TRAIT_AUDIO]: (nft: NFT) => cleanURL( getTrait(nft, 'Audio')),
+  [AudioUrlExtractorTypes.FIND_AUDIO_MEDIA]: (nft: NFT) => cleanURL(getMedia(nft, [MimeEnum.mp3, MimeEnum.wav, MimeEnum.xWav, MimeEnum.mp4, MimeEnum.quicktime]))
```
Comment (musnit) [inline src/types/fieldExtractor.ts:110]: can move the array into types/media as something like AudioAndVideoTypes ?

File: src/db/migrations/44-lens-approvals.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
+import { Knex } from 'knex';
+
+import { formatAddress } from '../../types/address';
+import { ChainId, getFactoryId } from '../../types/chain';
+import { CrdtOperation } from '../../types/message';
+import { Table } from '../db';
+
+const lensCurated = [
+  '0x727Ce17C09a74292Eb1594f70e8f048402b6DC02',
```
Comment (musnit) [inline src/db/migrations/44-lens-approvals.ts:9]: i see 39 tracks here but a bunch less on the preview @ https://spindexer-qa-build.spinamp.xyz/platform/lens

are some broken?

File: src/db/migrations/44-lens-approvals.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+      approved: true,
+      autoApprove: true,
+    },
+    operation: CrdtOperation.UPDATE,
+  }));
+
+  for (const message of messages) {
+    await knex(Table.seeds).insert(message);
+  }
+};
```
Comment (puffo) [inline src/db/migrations/44-lens-approvals.ts:56]: I think we should start to see if it's worthwhile trying out the new seed process which won't leave us with migrations to clean up later. We can also reflect on whether it is worth the extra steps or not :)

https://www.notion.so/Seed-process-for-Artist-Integrations-46e283d316644dfc92f413f44f8eae77

**Review by puffo — APPROVED** [review]

File: src/db/migrations/44-lens-approvals.ts (hunk 5 above)
Comment (musnit) [reply src/db/migrations/44-lens-approvals.ts:56]: @puffo yeah there's a task here assigned to u to migrate over to it when you're done with your current work: https://linear.app/spinamp/issue/SPI-676/deprecate-seed-migrations-properly

File: src/db/migrations/44-lens-approvals.ts (hunk 4 above)
Comment (Geimaj) [reply src/db/migrations/44-lens-approvals.ts]: Just had to remove a few duplicate values

**Pull request merged**